### DISCUSSION TITLE
Lettera di intento

### DISCUSSION THREAD

**Discussion by Annaanto** [fattureincloud/api#112] [Q&A] [answered]
Buongiorno,
sto cercando di creare cole le api 2.0 una fattura con lettera di intento
Ho passato le informazioni nell'oggetto entity

{
	"address_city" : "xxx",
	"address_postal_code" : "xxxxx",
	"address_province" : "xxx",
	"address_street" : "xxx",
	"country" : "Italia",
	"ei_code" : "xxx",
	"email" : "xxx",
	"has_intent_declaration" : true,
	"id" : 18542212,
	"intent_declaration_protocol_date" : "2021-12-22",
	"intent_declaration_protocol_number" : "xxx",
	"name" : "xxx",
	"phone" : "",
	"tax_code" : "xxx",
	"vat_number" : "xxx"
}

in anagrafica cliente c'è il flag sul campo Abilita lettera d'intento

Ma quando la fattura viene inviata in Fatture in Cloud negli attributi avanzati non n compare l'informazione sulla lettera di intento.
Cosa sbaglio?
Grazie,
Anna

**Comment by emajo** [comment, marked as answer]
Ciao @Annaanto per far sì che la lettera d'intento funzioni questi sono i passi da eseguire: 
- [creare un cliente](https://developers.fattureincloud.it/api-reference#post-/c/-company_id-/entities/clients) in anagrafica con i 3 campi relativi settati
- nella creazione della fattura ripassare tutti i campi del cliente creato (compreso id) all'interno del campo entity

A questo punto dovrebbe vedere i campi della lettera d'intento all'interno del campo ei_raw della fattura creata.

Mi faccia sapere se funziona,
grazie,
Emanuele

**Reply by nocodelab** [reply]
Ciao @emajo, anche io riscontro lo stesso ed identico problema.
Quando creo una fattura con articoli con iva 0% art 8c 633/72 non vengono aggiunti i riferimenti alla lettera d'intento.

L'anagrafica in questione ha i campi correttamente valorizzati:
<img width="265" alt="Screenshot 2023-01-17 at 18 03 40" src="https://user-images.githubusercontent.com/4540118/212964417-645b43c3-9a9e-4126-86bd-2e05fd776352.png">

Onde evitare errori manuali, ho anche provato ad inviare nel campo entity direttamente l'array ottenuto dalla chiamata alle Clients API.

Come posso risolvere?
Grazie

**Reply by valmoz** [reply]
Buongiorno @nocodelab 
Prima di tutto, ti suggerirei di postare il JSON completo della richiesta (ovviamente nascondendo i campi sensibili o usando valori fittizi), poiché dal solo array PHP della entity ci risulterà difficile capire il problema.

Secondariamente, per il supporto in fase di bug fix suggeriamo sempre di scrivere al supporto clienti invece che alla community, perché in quel caso potremo anche risalire al vostro company ID e andare a verificare il problema dai log.

A disposizione
Mauro

**Reply by nocodelab** [reply]
Ciao @valmoz, grazie per la tua risposta.
Preferisco continuare qui la conversazione in modo tale da poter aiutare altri sviluppatori nella mia stessa situazione in futuro.

Di seguito il JSON anonimizzato: https://pastebin.com/9F6vf7B9

**Reply by emajo** [reply]
Ciao @nocodelab, facendo una get del cliente i campi intent risultano valorizzati?

**Reply by nocodelab** [reply]
Certo @emajo, l'anagrafica è correttamente valorizzata nei tre campi richiesti:

         "has_intent_declaration":true,
         "intent_declaration_protocol_number":"XXXXX-000011",
         "intent_declaration_protocol_date":"XXXXXX-12-16",

**Reply by valmoz** [reply]
Buongiorno @nocodelab 
siamo riusciti a rintracciare i log delle sue chiamate, e a quanto ci risulta il campo entity non contiene l'ID del cliente.
Le chiederei di controllare che effettivamente il campo sia popolato.

Grazie
